### PR TITLE
Added option for linear lr schedule

### PR DESCRIPTION
**Pull request by DriessenA**
Adding the possibility to use a linear LR schedular with kwargs

### CODE REVIEW

File: cmonge/trainers/conditional_monge_trainer.py [hunk 1]
```diff
@@ -55,11 +55,18 @@ def init_model(self, datamodule: ConditionalDataModule):
...
+            lr_scheduler = optax.cosine_decay_schedule(
+                init_value=self.config.optim.lr,
+                decay_steps=self.num_train_iters,
+                alpha=1e-2,
+            )
+        elif self.config.lr_scheduler.name.lower() == "linear":
+            lr_scheduler = optax.linear_onecycle_schedule(
+                transition_steps=self.num_train_iters,
+                peak_value=self.config.optim.lr,
+                **self.config.lr_scheduler.kwags,
```
Comment (jannisborn) [inline cmonge/trainers/conditional_monge_trainer.py]: typo alarm: `kwargs` instead of `kwags` ?

File: cmonge/trainers/conditional_monge_trainer.py [hunk 2]
```diff
@@ -55,11 +55,18 @@ def init_model(self, datamodule: ConditionalDataModule):
 
         # setup optimizer and scheduler
         opt_fn = optim_factory[self.config.optim.name]
-        lr_scheduler = optax.cosine_decay_schedule(
-            init_value=self.config.optim.lr,
-            decay_steps=self.num_train_iters,
-            alpha=1e-2,
-        )
+        if self.config.lr_scheduler.name.lower() == "cosine":
```
Comment (jannisborn) [inline cmonge/trainers/conditional_monge_trainer.py]: you could use a`.get()..` here to ensure the default behavior is the cosine scheduler and guarantee that old configs will still run

**Review by jannisborn — CHANGES_REQUESTED** [review]

File: cmonge/trainers/conditional_monge_trainer.py (hunk 1 above)
Comment (DriessenA) [reply cmonge/trainers/conditional_monge_trainer.py]: Ah yeah whoops, typo

**Review by jannisborn — APPROVED** [review]
